nautobot/utilities/utils.py

```diff
@@ -187,6 +188,17 @@ def serialize_object_v2(obj):
     return data
 
 
+def model_to_json(obj, cls=None):
```

Comment:
False advertising! :-) This doesn't return a model's representation in JSON, it returns a Python dict that was constructed by way of rendering to JSON and parsing the JSON back into Python. Is that really the intent here? While I can see the use case for getting a JSON string representation of an ORM object (for logging, over-the-wire serialization, etc.) and that is why we have `serialize_object_v2`, I'm not sure I see the use case for converting an ORM object into a dict and then operating against the dict - what's the benefit of accessing `object["attribute"]` instead of `object.attribute`, etc.?

Reply:
I meant more, why not just `PolicyRule.objects.first().id`?

Reply:
Well, that is fine :) However, when it is nested, you have to re-invent what the serializer has done for you. 

Take a look at 
```
>>> PolicyRule.objects.first().source_zone.url
Traceback (most recent call last):
  File "<console>", line 1, in <module>
AttributeError: 'Zone' object has no attribute 'url'
>>> model_to_json(PolicyRule.objects.first())['source_zone']['url']
'/api/plugins/firewall/zone/89e4e313-e375-4990-999f-8b4e0b4925fd/'
>>> 
```

I find it difficult to find my way through, even here, I would create a for loop `PolicyRule.objects.first().source_zone.interfaces.all()` to understand the data, compared to just getting it.
```
>>> model_to_json(PolicyRule.objects.first())['source_zone']['interfaces']
[{'id': 'fa569847-25c4-4c28-9c44-fb31b3d2c8f8', 'url': '/api/dcim/interfaces/fa569847-25c4-4c28-9c44-fb31b3d2c8f8/', 'device': {'id': 'd77362cb-ccb4-411b-99a9-00eece0445b4', 'url': '/api/dcim/devices/d77362cb-ccb4-411b-99a9-00eece0445b4/', 'name': 'DFW-WAN00', 'display': 'DFW-WAN00'}, 'name': 'eth0', 'cable': None, 'display': 'eth0'}]
>>> PolicyRule.objects.first().source_zone.interfaces.all()
<RestrictedQuerySet [<Interface: eth0>]>
>>> 
```

Reply:
I suspect the utility of this function is not best solved in this thread, going to resolve in the mean time.